src/scene/scene.c

```diff
@@ -370,14 +374,23 @@ void sceneCheckPortals(struct Scene* scene) {
     int fireBlue = controllerActionGet(ControllerActionOpenPortal0);
     int fireOrange = controllerActionGet(ControllerActionOpenPortal1);
 
-    // this prevents the firing of portals after unpausing
-    if (scene->ignorePortalGun && (fireBlue || fireOrange)) {
-        fireBlue = 0;
-        fireOrange = 0;
-    } else {
-        scene->ignorePortalGun = 0;
+    // this prevents the firing of portals after unpausing or dropping an object
```

Comment:
It looks like both the pause and grab cases will work by just using the existing `controllerActionMuteActive()` function. It's used already to avoid this problem when loading levels.

Can you use `controllerActionMuteActive()` instead and remove the `ignorePortalGun`-related code? Then it will be cleaner and avoid duplication.

If in the future there is a need for `controllerActionMuteActive()` to mute more than just the portal actions, another function can be added in `controller_actions` specifically for portals. The logic around `gMutedActions` there is quite flexible.

Reply:
Makes you wonder why `ignorePortalGun` hasn't been layed off earlier.

This looks good. Yeah, if there are other actions added to the `HOLDABLE_ACTION()` macro, this will potentially need to be revised.